apps/assistants/tests/test_delete.py

```diff
@@ -28,36 +30,146 @@ def code_resource(assistant):
+
+        files_to_delete = list(_get_files_to_delete(tool_resource.assistant.team, tool_resource.id))
+        assert len(files_to_delete) == 0
+
+    def test_delete_openai_files_for_resource(self, code_resource):
+        all_files = list(code_resource.files.all())
+        assert all(f.external_id for f in all_files)
+        assert all(f.external_source for f in all_files)
+        client = Mock()
+        delete_openai_files_for_resource(client, code_resource.assistant.team, code_resource)
+
+        assert client.files.delete.call_count == 2
+        all_files = list(code_resource.files.all())
+        assert not any(f.external_id for f in all_files)
+        assert not any(f.external_source for f in all_files)
 
-    # only the second file should be deleted
-    files_to_delete = list(_get_files_to_delete(code_resource.assistant.team, code_resource.id))
-    assert len(files_to_delete) == 1
-    assert files_to_delete[0].id == all_files[1].id
 
-    files_to_delete = list(_get_files_to_delete(tool_resource.assistant.team, tool_resource.id))
-    assert len(files_to_delete) == 0
+# assistant.refresh_from_db()
 
 
 @pytest.mark.django_db()
-def test_delete_openai_files_for_resource(code_resource):
-    all_files = list(code_resource.files.all())
-    assert all(f.external_id for f in all_files)
-    assert all(f.external_source for f in all_files)
-    client = Mock()
-    delete_openai_files_for_resource(client, code_resource.assistant.team, code_resource)
-
-    assert client.files.delete.call_count == 2
-    all_files = list(code_resource.files.all())
-    assert not any(f.external_id for f in all_files)
-    assert not any(f.external_source for f in all_files)
+class TestAssistantArchival:
+    def test_archive_assistant(self):
+        assistant = OpenAiAssistantFactory()
+        assert assistant.is_archived is False
+        assistant.archive()
+        assert assistant.is_archived is True
+
+    @patch("apps.assistants.sync.push_assistant_to_openai", Mock())
+    def test_archive_assistant_succeeds_with_released_related_experiment(self):
+        exp_v1 = ExperimentFactory()
+        exp_v2 = exp_v1.create_new_version()
+        exp_v1.save()
+        assistant = OpenAiAssistantFactory()
+        exp_v2 = exp_v1.create_new_version()
+        exp_v2.assistant = assistant
+        exp_v2.save()
+        assert exp_v2.is_default_version is False
+        assert exp_v2.is_working_version is False
+        assistant.archive()
+        assistant.refresh_from_db()
+
+        assert assistant.is_archived is True  # archiving succeeded
+
+    @patch("apps.assistants.sync.push_assistant_to_openai", Mock())
+    def test_asistant_archive_blocked_by_working_related_experiment(self):
+        assistant = OpenAiAssistantFactory()
+        experiment = ExperimentFactory(assistant=assistant)
+        experiment.save()
+
+        assert experiment.is_working_version is True
+        assistant.archive()
```

Comment:
nit: you can also assert the return value:
```suggestion
        assert not assistant.archive()
```